### PR TITLE
Updated docs for CmfMenuBundle 2.0

### PR DESCRIPTION
**Pull request by WouterJ**
fixes https://github.com/symfony-cmf/symfony-cmf-docs/issues/679

### CODE REVIEW

File: book/structuring_content.rst [hunk 1]
```diff
@@ -105,36 +105,29 @@ rendered by the ``MenuFactory``.
 The Factory
 ...........
 
-The ``ContentAwareFactory`` is a ``FactoryInterface`` implementation, which
-generates the full ``MenuItem`` hierarchy from the provided MenuNode. The data
-generated this way is later used to generate the actual HTML representation of
-the menu.
+Menu factories generate the full ``MenuItem`` hierarchy from the provided
+MenuNode. The data generated this way is later used to generate the actual HTML
```
Comment (dbu) [inline book/structuring_content.rst]: should MenuNode be quoted as code as well?

File: bundles/menu/menu_factory.rst [hunk 2]
```diff
@@ -5,118 +5,109 @@ Menu Factory
...
-
-URL generation is absolute or relative, depending on the boolean value of the
-``routeAbsolute`` option.
+.. versionadded:: 2.0
+    Adding content support to the ``knp_menu.factory`` service was introduced
+    in CmfMenuBundle 2.0. Prior to 2.0, you had to use the
+    ``ContentAwareFactory`` class and ``cmf_menu.factory`` service.
+
+Most menu items will need a URL. By default, KnpMenu allows generating such
+route by specifying a URI or a Symfony route name.
```
Comment (dbu) [inline bundles/menu/menu_factory.rst]: such route? maybe "KnpMenu supports specifying the URL as an URI string or as a Symfony route name."

File: book/structuring_content.rst (hunk 1 above)
Comment (WouterJ) [inline book/structuring_content.rst]: changed it to "menu node", as it doesn't have to be a MenuNode class instance.

**Comment by WouterJ** [comment]
Thanks @dbu for the review. I've updated the PR accordingly

**Comment by dbu** [comment]
great!

**Pull request merged**